src/microbe_stage/Microbe.cs

```diff
@@ -588,7 +634,7 @@ public void ProcessEarlyAsync(float delta)
     public void ProcessSync(float delta)
     {
         // Updates the listener if this is the player owned microbe.
-        if (listener != null)
+        if (listener != null && IsInsideTree())
```

Comment:
When is IsInsideTree false when the microbe process system finds this in the tree?

Reply:
If I recall correctly, usually when some microbe is being engulfed (the player I think?).

Reply:
That should not be able to happen... I suppose if you move a microbe out of the scene tree in another microbe's process callback, it might happen in that case. Could you re-check this and add a comment about what situation this prevents?

Reply:
Okay I just remembered why this is. `IsInsideTree` is false when the player is digested by another cell and then removed (possibly because of a `DestroyDetachAndQueueFree`). Here's an error output:
```
E 0:01:05.214   get_global_transform: Condition "!is_inside_tree()" is true. Returned: Transform()
  <C++ Source>  scene/3d/spatial.cpp:272 @ get_global_transform()
  <Stack Trace> :0 @ void Godot.NativeCalls.godot_icall_0_22(IntPtr , IntPtr , Godot.Transform& )()
                Spatial.cs:322 @ Godot.Transform Godot.Spatial.GetGlobalTransform()()
                Spatial.cs:50 @ Godot.Transform Godot.Spatial.get_GlobalTransform()()
                Microbe.cs:650 @ void Microbe.ProcessSync(Single )()
                MicrobeSystem.cs:58 @ void MicrobeSystem.Process(Single )()
                MicrobeStage.cs:433 @ void MicrobeStage._Process(Single )()
```

Reply:
So the situation is that the microbe is queued for processing, but then another microbe object removes it from the game world tree? That's good enough explanation to me as long as this is commented in the code why the check is here.

Reply:
Or would it make more sense to have a DestroyAndQueueFree method? That way the microbe entity would be valid until the end of the current frame. Now thinking about that I'm kind of leaning towards preferring this. This might also become useful in other cases in the future.

Reply:
DestroyAndQueueFree is better. Added.